jina/executors/decorators.py

```diff
@@ -365,3 +372,107 @@ def arg_wrapper(*args, **kwargs):
         return _batching(func)
     else:
         return _batching
+
+def batching_ranker_input(func: Callable[[Any], np.ndarray] = None,
+                         batch_size: Union[int, Callable] = None,
+                         num_batch: Optional[int] = None,
+                         split_over_axis: int = 0,
+                         merge_over_axis: int = 0,
```

Comment:
since it is only for ranker, this will not be different than 0, so let's remove this. Also the `split_over_axis` (will it be different than 0?

Reply:
the same for `slice_on`?

Reply:
`slice_on` i feel can be a configurable parameter.

Reply:
at least `slice_on` should be set to the default that the `rankers` would use